src/App.tsx

```diff
@@ -1,26 +1,247 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoStatusFilter } from './types/TodoStatusFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAdding, setIsAdding] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filter, setFilter] = useState<TodoStatusFilter>(TodoStatusFilter.ALL);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    setErrorMessage(null);
+    setIsLoading(true);
+    inputRef.current?.focus();
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current);
+      timerRef.current = null;
+    }
+
+    if (errorMessage) {
+      timerRef.current = setTimeout(() => {
+        setErrorMessage(null);
+        timerRef.current = null;
+      }, 3000);
+    }
+
+    return () => {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current);
+        timerRef.current = null;
+      }
+    };
+  }, [errorMessage]);
+
+  function addNewTodo(title: string) {
```

Comment:
```suggestion
  function handleAddNewTodo(title: string) {
```
Be consistent with naming